src/topology/subset_properties.lean

```diff
@@ -1244,6 +1272,106 @@ begin
     { simpa using hs } }
 end
 
+/-- preconnected sets are either contained in, or disjoint to any given clopen set -/
+theorem subset_or_disjoint_of_clopen {α : Type*} [topological_space α] {s t : set α}
+  (h : is_preconnected t) (h1 : is_clopen s) : s ∩ t = ∅ ∨ t ⊆ s :=
+begin
+  apply classical.by_contradiction,
+  intro h2,
+  have h3 : s ∩ t ≠ ∅ ∧ sᶜ ∩ t ≠ ∅,
```

Comment:
Instead of the conjunction, state and prove the two facts separately: this will spare you a `split` in the proof, but also a `conv` below (and the awkward `h3.1` and `h3.2`). Also, you can state them directly with `nonempty`, as most lemmas use this form (with the current writing, you're going back and forth between one form or the other).

Also, your `apply` close to the end of the proof creates two different goals. Each of them should be enclosed in braces, to emphasize when a proof is done. 

With these modifications, this could become something like:
```lean
  by_contradiction h2,
  have h3 : (s ∩ t).nonempty := ne_empty_iff_nonempty.mp (mt or.inl h2),
  have h4 : (t ∩ sᶜ).nonempty,
  { apply set.inter_compl_nonempty_iff.2,
    push_neg at h2,
    exact h2.2 },
  rw [set.inter_comm] at h3,
  apply set.ne_empty_iff_nonempty.2 (h s sᶜ h1.1 (is_open_compl_iff.2 h1.2) _ h3 h4),
  { rw [set.inter_compl_self, set.inter_empty] },
  { rw [set.union_compl_self],
    exact set.subset_univ t },
```